language-support/hs/bindings/src/DA/Ledger.hs

```diff
@@ -59,5 +54,67 @@ getTransactionsPF lid party = do
     let req2 = GetTransactionsRequest lid now         Nothing    filter verbose
     stream <- getTransactions req1
     future <- getTransactions req2
-    past <- liftIO $ streamToList stream
+    past <- withTimeout $ liftIO $ streamToList stream
     return PastAndFuture { past, future }
+
+
+withGetTransactions
+    :: GetTransactionsRequest
+    -> (Stream [Transaction] -> LedgerService a)
+    -> LedgerService a
+withGetTransactions req act = do
+    stream <- getTransactions req
```

Comment:
This has the same issue that I described below for `withGetTransactionTrees` and it also looks like you should be able to reuse the logic (although once you turn it into a `bracket` it’s simple enough that I wouldn’t bother).

Reply:
done